src/main/java/teammates/common/datatransfer/FeedbackRubricQuestionDetails.java

```diff
@@ -51,32 +51,48 @@ public boolean extractQuestionDetails(
         
         int numOfRubricChoices = Integer.parseInt(numOfRubricChoicesString);
         int numOfRubricSubQuestions = Integer.parseInt(numOfRubricSubQuestionsString);
-        List<String> rubricChoices = new ArrayList<String>();
-        List<String> rubricSubQuestions = new ArrayList<String>();
-        List<List<String>> rubricDescriptions = new ArrayList<List<String>>();
+        List<String> rubricChoices = getRubricChoices(requestParameters, numOfRubricChoices);
+        List<String> rubricSubQuestions = getSubQuestions(requestParameters, numOfRubricSubQuestions);
+        List<List<String>> rubricDescriptions = getRubricQuestionDescriptions(requestParameters, 
+                                                                              numOfRubricChoices, numOfRubricSubQuestions);
         
-        int numActualChoices = 0;
-        int numActualSubQuestions = 0;
+        // Set details
+        setRubricQuestionDetails(rubricChoices.size(), rubricChoices, rubricSubQuestions.size(), 
+                                 rubricSubQuestions, rubricDescriptions);
```

Comment:
Better yet, just make `setRubricQuestionDetails` take in 3 parameters. I believe you can get the sizes for the passed in lists in the function itself.